### PR TITLE
Testing/update navigation and user messages

### PR DESCRIPTION
**Pull request by elizhahn**
## What does this PR do (summary of changes)?
 - This updates the testing suite to include: 
     -  testing for server error message
     - testing for search bar messages
     - updates testing for navigation to include navigating to/from home
     - updates testing for home page elements and nav bar elements
     - updates testing for song book components to dynamically check all song cards for correct information
## How should it be tested?
 - fetch the branch and run the testing suite
 - check for syntax errors and formatting issues
## Future Iterations:
 - Am I missing any testing?

### CODE REVIEW

File: cypress/integration/carry_okay_spec.js [hunk 1]
```diff
@@ -1,89 +1,102 @@
...
+      cy.get('[data-cy=server-error-msg]').contains('Oops, our system seems to be down... how embarassing. Try again in a few moments!');
+  
+    });
+  });
+  
  beforeEach(() => {
-  cy.intercept( 'https://carryokay-server.herokuapp.com/genres', {fixture:"genre_data.json"})
-  cy.intercept('https://carryokay-server.herokuapp.com/songs', {fixture:"song_data.json"})
+  cy.intercept( '/genres', {fixture:"genre_data.json"})
+  cy.intercept('/songs', {fixture:"song_data.json"})
```
Comment (monshan) [inline cypress/integration/carry_okay_spec.js:22]: nice work with the shorthand

File: cypress/integration/carry_okay_spec.js [hunk 2]
```diff
@@ -1,89 +1,102 @@
...
-
   });
 
   describe('Navigation', () => {
 
     it('should be able to navigate to My Songs view from Home view', () => {
       cy.get('[data-cy=my-songs-nav]').click();
-      cy.get("CarryOkay").should('not.exist');
-      cy.get("Hello friend 😬").should('not.exist');
-      cy.url().should("eq", "http://localhost:3000/mysongs")
```
Comment (monshan) [inline cypress/integration/carry_okay_spec.js:29]: I'm assuming this means we are sticking with single quotes for the testing file, let me know if this is how we should be approaching the js files as well

File: cypress/integration/carry_okay_spec.js [hunk 3]
```diff
@@ -93,94 +106,96 @@ describe('CarryOkay', () => {
...
-        .and("contain", "Radiohead")
-        .and("contain","Electronica")
-        .and("contain","Rock")
+    it('should allow a user to add a song to their library', () => {
+      cy.get('[data-cy=song-card-btn]').first().click();
+      cy.get('[data-cy=my-songs-nav]').click();
+      cy.get('[data-cy=song-card]').should('contain', 'Jigsaw falling into place')
+          .and('contain', 'Radiohead')
+          .and('contain','Electronica')
+          .and('contain','Rock');
```
Comment (monshan) [inline cypress/integration/carry_okay_spec.js:172]: small, but are we missing spacing on these two lines or the other way around?

File: cypress/integration/carry_okay_spec.js (hunk 2 above)
Comment (elizhahn) [reply cypress/integration/carry_okay_spec.js:29]: I noticed most of it was single quotes, so I went with that standard, in the .js files we might wanna decide on that. I think it's single quotes but we use double quotes in the JSX like how you use double quotes in HTML?

File: cypress/integration/carry_okay_spec.js (hunk 3 above)
Comment (elizhahn) [reply cypress/integration/carry_okay_spec.js:172]: I noticed it was tabbed over to the right an extra space is that what you mean?

File: cypress/integration/carry_okay_spec.js [hunk 4]
```diff
@@ -93,94 +106,96 @@ describe('CarryOkay', () => {
...
+          });
+        });
+      cy.fixture('/genre_data.json').then((data) => {
+        data.forEach(genre => {
+          const genres = Object.entries(genre)
+          genres.forEach(entry => {
+            if(entry.includes(true)) {
+              cy.get('[data-cy=song-card]').contains(entry[0]);
+            }
+          });
```
Comment (monshan) [inline cypress/integration/carry_okay_spec.js:144]: these fixtures looping through with a forEach essentially check every entry, perhaps the first more than the second but potentially shorten it for optimization if a preference

Comment (elizhahn) [reply cypress/integration/carry_okay_spec.js:144]: What do you mean by shorten it?

File: cypress/integration/carry_okay_spec.js (hunk 3 above)
Comment (monshan) [reply cypress/integration/carry_okay_spec.js:172]: line 170 'Radiohead' has some spacing right before it vs 171-172 don't

File: cypress/integration/carry_okay_spec.js (hunk 4 above)
Comment (monshan) [reply cypress/integration/carry_okay_spec.js:144]: instead of a ForEach maybe a while or do/while loop limiting it to 305 iterations

Comment (elizhahn) [reply cypress/integration/carry_okay_spec.js:144]: ok, we can look at it together, I don't have much experience with the do..while loop

**Comment by m1073496** [comment]
[Add testing for user messages](https://trello.com/c/XYv7nZOb/90-add-testing-for-user-messages)

**Pull request merged**